File: backend/src/main/java/com/example/knowledgehub/backend/resource/LearningResource.java
package com.example.knowledgehub.backend.resource;

import com.example.knowledgehub.backend.category.Category;
import jakarta.persistence.*;

@Entity
@Table(name = "learning_resources")
public class LearningResource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String url;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    public LearningResource() {
    }

    public LearningResource(String name, String description, String url, Category category) {
        this.name = name;
        this.description = description;
        this.url = url;
        this.category = category;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }
}

File: backend/src/main/java/com/example/knowledgehub/backend/resource/LearningResource.kt
package com.example.knowledgehub.backend.resource

import com.example.knowledgehub.backend.category.Category
import jakarta.persistence.*

@Entity
@Table(name = "learning_resources")
class LearningResource(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    var category: Category,

    @Column(nullable = false)
    var title: String,

    @Column(nullable = false, columnDefinition = "TEXT")
    var shortDescription: String,

    @Column(nullable = false)
    var resourceUrl: String,

    var iconUrl: String? = null,

    var orderIndex: Int? = null,
)
